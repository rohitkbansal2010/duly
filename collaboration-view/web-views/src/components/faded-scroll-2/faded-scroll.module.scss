@import "~@styles/main.scss";
@import "~@styles/colors.scss";

.scrollWrapper {
  position: relative;

  .scrollContent {
    @include invisible-scroll;
    height: 100%;
    -ms-touch-action: pan-y;
    touch-action: pan-y;
  }

  &.oversize {
    .topScrollFade {
      top: -1px;
      right: -1px;
      left: -1px;
    }

    .bottomScrollFade {
      right: -1px;
      bottom: -1px;
      left: -1px;
    }
  }
}

.topScrollFade {
  @include invisible-scroll-top-fade;
  max-height: 25%;
}

.top0 {
  opacity: 0;
}

.top25 {
  opacity: 0.25;
}

.top50 {
  opacity: 0.5;
}

.top75 {
  opacity: 0.75;
}

.top100 {
  opacity: 1;
}

.bottomScrollFade {
  @include invisible-scroll-bottom-fade;
  max-height: 25%;

  &.bottom-white {
    background-image: linear-gradient(transparent, $white);
  }
}

.bottom0 {
  opacity: 0;
}

.bottom25 {
  opacity: 0.25;
}

.bottom50 {
  opacity: 0.5;
}

.bottom75 {
  opacity: 0.75;
}

.botom100 {
  opacity: 1;
}
