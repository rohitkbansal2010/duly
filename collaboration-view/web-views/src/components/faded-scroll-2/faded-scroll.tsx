import { PropsWithChildren, useRef } from 'react';

import { useScrollFades } from '@hooks';

import styles from './faded-scroll.module.scss';

type Props = PropsWithChildren<{
    height: string;
    className?: string;
    view?: 'default' | 'bottom-white';
    oversize?: boolean
}>

export const FadedScroll = (props: Props) => {
  const {
    children, height, className, view = 'default', oversize = false,
  } = props;
  const containerRef = useRef<HTMLDivElement>(null);

  const { topFadeClassname, bottomFadeClassname } = useScrollFades(containerRef);

  const topScrollFade = ([] as string[])
    .concat(styles.topScrollFade)
    .concat(styles[topFadeClassname])
    .concat(styles[view])
    .join(' ');

  const bottomScrollFade = ([] as string[])
    .concat(styles.bottomScrollFade)
    .concat(styles[bottomFadeClassname])
    .concat(styles[view])
    .join(' ');

  const wrapperClasses = [ styles.scrollWrapper ];
  className && wrapperClasses.push(className);
  oversize && wrapperClasses.push(styles.oversize);

  return (
    <div className={wrapperClasses.join(' ')} style={{ height }}>
      <div className={topScrollFade} />
      <div ref={containerRef} className={styles.scrollContent}>
        {children}
      </div>
      <div className={bottomScrollFade} />
    </div>
  );
};
