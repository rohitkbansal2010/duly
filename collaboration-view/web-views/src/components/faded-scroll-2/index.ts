export { FadedScroll } from './faded-scroll';
