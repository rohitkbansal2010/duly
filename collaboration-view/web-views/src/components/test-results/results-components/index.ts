export { ResultsComponents } from './results-components';
