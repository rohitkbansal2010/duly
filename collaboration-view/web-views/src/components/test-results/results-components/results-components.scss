@import "~@styles/typography";
@import "~@styles/colors";
@import "~@styles/main.scss";

@mixin name {
  width: 28%;
}

@mixin value {
  width: 20%;
}

@mixin range {
  width: 20%;
}

@mixin additional {
  width: 32%;
}

@mixin header-item {
  padding: 0.75rem 1.25rem;
}

@mixin component-item {
  display: flex;
  align-items: center;
  padding: 1.25rem;
  word-break: break-word;
}

@mixin border-right {
  border-right: 0.0625rem solid $gray;
}

.results {
  &Components {
    display: flex;
    flex-direction: column;
    flex-grow: 1;
    min-height: 0;
    margin-top: 1.25rem;

    &Header {
      @include inter-text-xs-bold;

      display: flex;
      font-weight: 700;
      font-size: var(--font-size-14px-4px);
      text-transform: uppercase;
      background: rgba($dupage-light-blue, 0.1);
      border: 0.0625rem solid $gray;
      border-radius: 0.5rem 0.5rem 0 0;

      &Name {
        @include name;
        @include header-item;
        @include border-right;
      }

      &Value {
        @include value;
        @include header-item;
        @include border-right;
      }

      &Range {
        @include range;
        @include header-item;
        @include border-right;
      }

      &Additional {
        @include additional;
        @include header-item;
      }
    }

    &List {
      position: relative;
      flex-grow: 1;
      min-height: 0;
    }
  }
}
