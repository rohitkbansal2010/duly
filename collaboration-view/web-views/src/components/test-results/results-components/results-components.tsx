import noop from 'lodash/noop';
import { useCallback } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { FadedScroll } from '@components/faded-scroll-2';
import { TEST_RESULTS_ACCORDION_TYPES } from '@constants';
import { TestResultsAccordions } from '@enums';
import { addAccordion, deleteAccordion } from '@redux/actions';
import { RootState } from '@redux/reducers';
import { TestResult } from '@types';

import { ResultsItem } from '../results-item';

import styles from './results-components.scss';

type ResultsComponentsProps = {
  results?: TestResult[];
  currentTestResultsId?: string;
}

export const ResultsComponents = ({ currentTestResultsId = '', results }: ResultsComponentsProps) => {
  const dispatch = useDispatch();
  const accordions = useSelector(({ CURRENT_APPOINTMENT }: RootState) =>
    CURRENT_APPOINTMENT.accordions);

  const handleClick = useCallback((id: string, type: TestResultsAccordions) =>
    () => {
      const currentId = `${id}-${TEST_RESULTS_ACCORDION_TYPES[type].current}`;
      const otherId = `${id}-${TEST_RESULTS_ACCORDION_TYPES[type].other}`;

      dispatch(
        accordions[currentTestResultsId]?.includes(currentId)
          ? deleteAccordion(currentTestResultsId, currentId)
          : addAccordion(currentTestResultsId, currentId)
      );

      accordions[currentTestResultsId]?.includes(otherId) &&
        dispatch(deleteAccordion(currentTestResultsId, otherId));
    }, [ dispatch, accordions, currentTestResultsId ]);

  const getActiveKey = useCallback((id: string) =>
    accordions[currentTestResultsId]?.find((accordionId: string) =>
      accordionId === `${id}-${TestResultsAccordions.NOTES}` ||
        accordionId === `${id}-${TestResultsAccordions.PAST_RESULTS}`)
      || null, [ accordions, currentTestResultsId ]);

  return (
    <>
      {!!results?.length && (
      <div className={styles.resultsComponents}>
        <div className={styles.resultsComponentsHeader}>
          <div className={styles.resultsComponentsHeaderName}>components</div>
          <div className={styles.resultsComponentsHeaderValue}>your value</div>
          <div className={styles.resultsComponentsHeaderRange}>standard range</div>
          <div className={styles.resultsComponentsHeaderAdditional} />
        </div>
        <div className={styles.resultsComponentsList}>
          <FadedScroll height="100%" view="bottom-white">
            {results.map(({ id, ...rest }) =>
              (
                <ResultsItem
                  key={id}
                  eventKey={id}
                  activeKey={getActiveKey(id) as string}
                  onClickNotes={handleClick(id, TestResultsAccordions.NOTES)}
                  onClickPastResults={noop}
                  {...rest}
                />
              ))}
          </FadedScroll>
        </div>
      </div>
      )}
    </>
  );
};
