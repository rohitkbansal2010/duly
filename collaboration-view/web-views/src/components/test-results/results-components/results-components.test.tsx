import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as redux from 'react-redux';

import { FadedScroll } from '@components/faded-scroll-2';
import { testReportsResultMockData } from '@mock-data';

import { ResultsComponents } from './results-components';

describe('ResultsComponents', () => {
  let component: ReturnType<typeof shallow>;

  const {
    results,
    id,
  } = testReportsResultMockData;

  const mockProps = {
    results,
    currentTestResultsId: id,
  };

  beforeEach(() => {
    jest
      .spyOn(redux, 'useDispatch')
      .mockReturnValue(jest.fn());

    jest
      .spyOn(redux, 'useSelector')
      .mockReturnValue({ accordions: {} });

    component = shallow(<ResultsComponents {...mockProps} />);
  });

  it('component should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('component should contain invisible scroll', () => {
    expect(component.find(FadedScroll)).toHaveLength(1);
  });

  it('component should contain right titles', () => {
    expect(component.text().includes('component')).toBeTruthy();
    expect(component.text().includes('your value')).toBeTruthy();
    expect(component.text().includes('standard range')).toBeTruthy();
  });

  it('component should work with empty data', () => {
    component = shallow(<ResultsComponents />);

    expect(component.text().includes('undefined')).toBeFalsy();
    expect(toJson(component)).toMatchSnapshot();
  });
});
