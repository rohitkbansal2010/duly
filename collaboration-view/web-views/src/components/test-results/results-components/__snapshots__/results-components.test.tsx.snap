// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsComponents component should match snapshot 1`] = `
<Fragment>
  <div
    className="resultsComponents"
  >
    <div
      className="resultsComponentsHeader"
    >
      <div
        className="resultsComponentsHeaderName"
      >
        components
      </div>
      <div
        className="resultsComponentsHeaderValue"
      >
        your value
      </div>
      <div
        className="resultsComponentsHeaderRange"
      >
        standard range
      </div>
      <div
        className="resultsComponentsHeaderAdditional"
      />
    </div>
    <div
      className="resultsComponentsList"
    >
      <FadedScroll
        height="100%"
        view="bottom-white"
      >
        <ResultsItem
          activeKey={null}
          componentName="Cholesterol, Total"
          eventKey="06cd80cf-7230-4e53-bdd1-6956402b2ef8"
          isAbnormalResult={false}
          key="06cd80cf-7230-4e53-bdd1-6956402b2ef8"
          measurement={
            Object {
              "unit": "mg/dL",
              "value": 150,
            }
          }
          notes={
            Array [
              "Desirable  <200 mg/dL\\\\r\\\\nBorderline  200-239 mg/dL\\\\r\\\\nHigh      >=240 mg/dL\\\\r\\\\n\\\\r\\\\n",
            ]
          }
          onClickNotes={[Function]}
          onClickPastResults={[Function]}
          referenceRange={
            Object {
              "text": "<200",
            }
          }
        />
        <ResultsItem
          activeKey={null}
          componentName="LDL Cholesterol"
          eventKey="9bea73b9-d173-4cc4-b67b-fb43971557e4"
          isAbnormalResult={true}
          key="9bea73b9-d173-4cc4-b67b-fb43971557e4"
          measurement={
            Object {
              "unit": "mg/dL",
              "value": 117,
            }
          }
          notes={
            Array [
              "Desirable <100 mg/dL \\\\r\\\\nBorderline 100-129 mg/dL \\\\r\\\\nHigh     >=130mg/dL\\\\r\\\\n\\\\r\\\\n",
            ]
          }
          onClickNotes={[Function]}
          onClickPastResults={[Function]}
          referenceRange={
            Object {
              "text": "<100",
            }
          }
        />
        <ResultsItem
          activeKey={null}
          componentName="Sodium"
          eventKey="c1f12ce2-323e-4c7b-b3e0-73183ad58330"
          isAbnormalResult={false}
          key="c1f12ce2-323e-4c7b-b3e0-73183ad58330"
          measurement={
            Object {
              "unit": "mmol/L",
              "value": 138,
            }
          }
          onClickNotes={[Function]}
          onClickPastResults={[Function]}
          referenceRange={
            Object {
              "high": Object {
                "unit": "mmol/L",
                "value": 145,
              },
              "low": Object {
                "unit": "mmol/L",
                "value": 136,
              },
              "text": "136 - 145 mmol/L",
            }
          }
        />
      </FadedScroll>
    </div>
  </div>
</Fragment>
`;

exports[`ResultsComponents component should work with empty data 1`] = `<Fragment />`;
